components/butterfly/src/rumor.rs

```diff
@@ -55,6 +56,35 @@ lazy_static! {
                                   &["rumor"]).unwrap();
 }
 
+#[derive(Clone, Debug, Serialize)]
+pub struct RumorExpiration(DateTime<Utc>);
+
+impl RumorExpiration {
+    // Some rumors we don't want to ever have naturally age out. We only want them
+    // to expire when there is a new rumor to take their place (e.g. a new election). When that
+    // type of trigger event happens, we will manually set the expiration date to a short time in
+    // the future. Until then, though, we don't want this rumor to go away on its own, so we set
+    // the expiration date to 100 years in the future, which is effectively forever for our
+    // purposes. This is far more convenient and natural to work with than having to deal with
+    // Option<DateTime<Utc>>.
```

Comment:
This was to deal with each rumor having `Option<RumorExpiration>` in its struct instead of just `RumorExpiration`.

Having a concept of `forever` solves the problem of having to check/work around the case where we don't want a rumor to expire, e.g. `Departure`.  Without this mechanism, we need to make the `expiration` field on each rumor an `Option<RumorExpiration>` so that e.g. `Departure` can have `None`.

In practice, this makes all of the code that deals with rumor expiration more cumbersome, because you're always having to check for `None`.  Setting it to a really far away time ensures that it won't expire (at least not in any of our lifetimes) and also lets us avoid awkward checks for `Option`s everywhere.